jsonschema/_typing.py

```diff
@@ -0,0 +1,14 @@
+"""
+Type aliases and utilities to help manage `typing` usage and type annotations.
+"""
+
+from collections.abc import Mapping, Sequence
+import typing
+
+Schema = Mapping[str, typing.Any]
+
+JsonObject = Mapping[str, typing.Any]
+
+JsonValue = typing.Union[
```

Comment:
This is incomplete -- someone can come along and invent validator callbacks which only work if someone deserialized into `Decimal`, and as long as they ensure they've done so in their loading of JSON, their code works. In other words, I'm somewhat afraid this is really just `Any` in "real life".

Reply:
I have to think about what I think honestly.

To me, all I am "convinced" is useful of at the minute (and I'm quite convinced it's very useful indeed) is that typing forces documentation to have correctly documented types. That's the reason I'm quite picky about making sure that things which work "accidentally" because we abuse some type internally do not in fact get represented as the type -- because later down the line I will *not* support those internal abuses if we need to change them for whatever reason. They're not part of the public API, they don't have stability guarantees.

Here, the opposite is true -- *not* documenting the type here actually means "regressing" something that is indeed public API, and which I very much do intend folks to rely on (by which I don't mean I have no reservations that it will bite me in the ass, but I do mean it's something I think people *should* consider public API at this point, type annotation or not, and I don't intend to break it).

And I know we don't actually regress, since I'm pretty sure there are tests covering the `Any` behavior at least in part -- but if the goal (at least the goal in my head) is "be able to say every object in `jsonschema` has its type documented explicitly", it complicates that goal if the documentation again needs tweaking to *broaden* types into their real public API.

(So... "I don't know yet" :D)